wallet/src/de/schildbach/wallet/service/CoinJoinService.kt

```diff
@@ -377,4 +417,12 @@ class CoinJoinMixingService @Inject constructor(
     // TODO: private val _progressFlow = MutableStateFlow(0.00)
     // TODO: override suspend fun getMixingProgress(): Flow<Double + other data> = _progressFlow
     // TODO: suspend fun setProgress(progress: Double) = _progressFlow.emit(progress)
+
+    /** clear previous state */
+    private suspend fun clear() {
+        exception = null
+        mixingStatus = MixingStatus.NOT_STARTED
+        if (!mixingMutex.isLocked)
+            mixingMutex.lock()
+    }
```

Comment:
This will allow CoinJoinService to perform another mixing job after finishing the previous job.